### PR TITLE
[docs] RUM JS documentation restructure

### PR DESCRIPTION
**Pull request by bmorelli25**
**WIP** For https://github.com/elastic/apm-agent-rum-js/issues/511. Demo available **[here](http://apm-agent-rum-js_512.docs-preview.app.elstc.co/diff)**.

- [x] Use shared version attributes: https://github.com/elastic/apm/issues/150
- [x] Add a section on how the Agent does instrumentation: https://github.com/elastic/apm/issues/146
- [x] Rename the APM UI to "APM app": https://github.com/elastic/apm-dev/issues/567
- [x] Use `apm-app-ref`: https://github.com/elastic/docs/pull/1276
- [x] Add placeholder .asciidoc file for release notes - https://github.com/elastic/docs/pull/1502
- [x] Updates to the new documentation structure outlined [here](https://docs.google.com/spreadsheets/d/1P53LbNY7jFSLLJhOKw-nmDenQbrIFlY4EEFJGqT1kmY/edit#gid=0)
- [ ] ~Sync Central config docs: https://github.com/elastic/apm-agent-dotnet/pull/497#discussion_r326303325~
- [ ] ~Top level heading and doc sync for log correlation: https://github.com/elastic/apm/issues/149~

### CODE REVIEW

**Comment by bmorelli25** [comment]
@jahtalab @vigneshshanmugam -- Still a work in progress, but here's what I'm thinking so far for the intro / set up / supported technology sections of the documentation.


## Introduction
There was a lot of great information about how the agent works that was burried under the supported-tech header. I think it makes more sense to present this information at the beginning of the documentation. Any feedback as to how to improve this would be awesome.
Screen Shot 2019-12-03 at 10 15 36 AM

## Set up
Three steps, with the pages for installing the Agent and setting up CORS nested underneath. Is "install the agent" the correct terminology to use here?
Screen Shot 2019-12-03 at 10 15 49 AM

## Supported technologies
Screen Shot 2019-12-03 at 10 15 55 AM

**Comment by vigneshshanmugam** [comment]
@bmorelli25 This looks really good, thanks for initiating this Brandon 🎉. One feedback from my side would be to move top level docs like custom transaction, distributed tracing, opentracing in to advanced sections to keep the overall index page simple. We can also do this on next iteration.

**Comment by bmorelli25** [comment]
Great, thanks for the feedback! I'm still planning on attacking the rest of the documentation in this PR, just had to switch gears yesterday and didn't have enough time to finish.

**Comment by bmorelli25** [comment]
Alrighty, I've opened up the release note changes in a separate PR (https://github.com/elastic/apm-agent-rum-js/pull/516).

> One feedback from my side would be to move top level docs like custom transaction, distributed tracing, opentracing in to advanced sections to keep the overall index page simple.

Sooo let's chat about this.
- **Custom transaction** is already under the advanced section, which I agree with.
- I'm hesitant to hide **distributed tracing** under the advanced section, because to me, that seems like a huge use case for the RUM Agent. I don't want to make it difficult to find.
- As for **Opentracing**, we [left it as a top level heading](https://docs.google.com/spreadsheets/d/1P53LbNY7jFSLLJhOKw-nmDenQbrIFlY4EEFJGqT1kmY/edit#gid=0) for all of the other Agents, so I'd like to match that format.

File: docs/intro.asciidoc [hunk 1]
```diff
@@ -1,159 +1,36 @@
 [[intro]]
 == Introduction
 
-Welcome to the APM Real User Monitoring (RUM) JavaScript Agent documentation.
+The Elastic APM RUM JavaScript Agent provides detailed performance metrics and error tracking of your web applications.
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/intro.asciidoc]: YES! Good catch

**Pull request merged**